sql/src/test/java/io/crate/breaker/RowAccountingTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.breaker;
+
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.testing.RowGenerator;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.breaker.MemoryCircuitBreaker;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.junit.Test;
+
+import java.util.Collections;
+
+public class RowAccountingTest extends CrateUnitTest {
+
+    @Test
+    public void testCircuitBreakingWorks() throws Exception {
+        RowAccounting rowAccounting = new RowAccounting(Collections.singletonList(DataTypes.INTEGER),
+            new RamAccountingContext(
+                "test",
+                new MemoryCircuitBreaker(
+                    new ByteSizeValue(10, ByteSizeUnit.BYTES), 1.01, Loggers.getLogger(RowAccountingTest.class))
+            ));
+
+        expectedException.expect(CircuitBreakingException.class);
+        // internal flush buffer is 2MB, so it doesn't immediately trip after the first row
```

Comment:
you could change ( and reset afterwards) the flush buffer size... not sure if its worth, should be pretty fast already or?